client/src/components/Today/use-detailed-activity-modal.ts

```diff
@@ -0,0 +1,17 @@
+import modalIds from "@/lib/modal-ids";
+import { useModalState } from "@/lib/state/modal-state";
+import type { ActivityWithIds } from "@/types/server/activity.types";
+
+export function useDetailedActivityModal({ activity }: { activity: ActivityWithIds }) {
+	const { setModalState } = useModalState(modalIds.detailedActivity);
+
+	function openDetailedActivityModal() {
+		setModalState(() => ({
+			isOpen: true,
+			itemId: activity.activity_id,
+			itemType: "activity",
+		}));
+	}
+
+	return { openDetailedActivityModal };
```

Comment:
Maybe just call it `openModal` since it's already clear from the name of the hook which modal it's referring to.